quickjs.c

```diff
@@ -40396,6 +40429,126 @@ static JSValue js_iterator_proto_set_toStringTag(JSContext *ctx, JSValue this_va
     return JS_UNDEFINED;
 }
 
+typedef struct JSIteratorHelperData {
+    JSIteratorHelperKindEnum kind; 
+    JSValue obj;
+    JSValue next;
+    JSValue func; // predicate (filter) or mapper (flatMap, map)
+    uint32_t limit; // (drop, take)
+} JSIteratorHelperData;
```

Comment:
2 of the other failing tests involve keeping some state so I'll likely need to add a flag indicating if we are currently inside next() and another for knowing if the underlying iterator is exhausted so we can stop forwarding return() calls.